Source/displayinfo/include/ExtendedDisplayIdentification.h

```diff
@@ -124,22 +187,341 @@ namespace Plugin {
         };
 
         class CEA {
+        public:
+            static constexpr uint8_t extension_tag = 0x02;
+        public:
+            class DataBlockIterator {
+            public:
+                DataBlockIterator() = delete;
+                DataBlockIterator(const Buffer& rhs, uint8_t dtdBegin)
+                    : _segment(rhs)
+                    , _index(4)
+                    , _dtdBegin(dtdBegin)
+                    , _reset(true) {
+                        ASSERT(_index >= 4);
+                }
+                DataBlockIterator(const DataBlockIterator& copy)
+                    : _segment(copy._segment)
+                    , _index(copy._index)
+                    , _dtdBegin(copy._dtdBegin)
+                    , _reset(true) {
+                        ASSERT(_index >= 4);
+                }
+                ~DataBlockIterator() = default;
+
+                DataBlockIterator& operator= (const DataBlockIterator& rhs) {
```

Comment:
Ni implementation for the assignment operator? Should you not copy at least the buffer and set the index to teh beginning, or also copy ?

Reply:
Yes, you are right. I forgot about that. Fixed by providing assignment operator for DataBlockIterator, Iterator and Buffer